packages/datepicker/src/react/text-input-field.tsx

```diff
@@ -0,0 +1,58 @@
+import Field from '@pluralsight/ps-design-system-field'
+import { CalendarIcon } from '@pluralsight/ps-design-system-icon'
+import { useUniqueId } from '@pluralsight/ps-design-system-util'
+import React, { ComponentProps, FC } from 'react'
+
+interface TextInputFieldProps extends ComponentProps<typeof Field> {
+  onChange?: React.ChangeEventHandler<HTMLInputElement>
+  placeholder?: string
+  value?: React.ReactText
+}
+
+export const TextInputField: FC<TextInputFieldProps> = props => {
+  let {
+    disabled,
+    label,
+    onChange,
+    onClick,
+    placeholder,
+    prefix,
+    subLabel,
+    value,
+    ...rest
+  } = props
+  const labelId = useUniqueId('text-input__label-')
+  const inputId = useUniqueId('text-input__input-')
+
+  if (!prefix) {
+    prefix = <CalendarIcon onClick={onClick} style={{ cursor: 'pointer' }} />
+  }
+  if (typeof label === 'string') {
```

Comment:
should this be memoized like i've done in other components https://github.com/pluralsight/design-system/blob/master/packages/multiselect/src/react/index.tsx#L203

not sure but might want to look into it. i've memoized because of two issues i'd encountered
1. unnecessary renders
2. refs issues that i don't think would matter here